src/mds/FSMap.h

```diff
@@ -447,6 +447,29 @@ class FSMap {
     fs->mds_map.epoch = epoch;
   }
 
+  void rename_fs(fs_cluster_id_t fscid, std::string& new_fs_name)
+  {
+    auto& fs = filesystems.at(fscid);
+    fs->mds_map.set_fs_name(new_fs_name);
+    fs->mds_map.epoch = epoch;
+  }
```

Comment:
@vshankar @batrick 
Is it worth it to have method `rename_fs()`? Is it better than creating and passing a lambda function to `modify_filesystem()`? Or is it rather misleading since it renames FS but doesn't take care of any other details. For example,  setting FS name in OSD pool application key-value pair.

Reply:
won't this change the uniformity in code?

Reply:
Its not misleading. However, the `modify_filesystem` contruct should be used since it takes care of updating the MDSMap epoch.

Reply:
Venky's point is right.

Reply:
@rishabh-d-dave Let's use the modify_filesystem() construct so that any changes that requires updating things in mdsmap is done in one place and is doesn't get missed.

Reply:
Got it.

To more sure I have understood what you mean -- you mean someone in future might add a new method that is similar to `rename_fs()` but the author might forget to add the code to update the MDSMap epoch. Therefore, it's better to avoid adding new methods, like `rename_fs()` and use the `modify_filesystem()` instead. Correct?